### ISSUE TITLE
[Shopping Cart Elite]

### ISSUE THREAD

**Issue by stephen485** [PipedreamHQ/pipedream#7459]
**SCE API** https://api.shoppingcartelite.com/master/sceapi.asmx

**[Link to developer documentation](https://api.shoppingcartelite.com/master/sceapi.asmx)**

**Is lack of support preventing you from building workflows, or do you have a workaround?**

To build a workflow, we need SCE support to avoid the workaround of using frequent CSV exports.

**Are there specific actions, or triggers, you'd like to see for this app? Please let us know here or use the Action and Trigger issue templates to open requests for each!**

Get Autoship Table from SCE **Action**
Create new table **Action**
Make Email column primary key **Action**
New row added to table **Trigger**
Table row updated **Trigger**
Find Diff **Action*
Highlight Diffs in new table **Action**
Add Changed rows prior value as reference **Action**
Send updated table to SmartSuite **Action**

**Comment by sergio-eliot-rodriguez** [comment]
This uses a SOAP Web Services API

**Comment by stephen485** [comment]
Does working with SOAP pose any problems, or is it just less flexible?
SCE claims to use REST API; perhaps they offer 2 APIs?



This uses a SOAP Web Services API

**Comment by sergio-eliot-rodriguez** [comment]
Yes, currently SOAP APIs aren't supported. Can you share a link for SCE's REST API documentation? or point me within the docs ytou shared? i.e. asmx are a native .NET file extension that they use for SOAP based services.

**Comment by sergio-eliot-rodriguez** [comment]
Thanks for your reply.

I found that the text you are sharing was taking from this article:

https://www.shoppingcartelite.com/articles/api-development-tools

Funny enough, ASMX is:

> ASMX provides the ability to build web services that send messages using the Simple Object Access Protocol (SOAP). SOAP is a platform-independent and language-independent protocol for building and accessing web services

And if you click on any of the methods listed in the asmx file, such as AccountSearch
https://api.shoppingcartelite.com/master/sceapi.asmx?op=AccountSearch
you get

SOAP 1.1
The following is a sample SOAP 1.1 request and response. The placeholders shown need to be replaced with actual values.

> POST /master/sceapi.asmx HTTP/1.1
> Host: api.shoppingcartelite.com
> Content-Type: text/xml; charset=utf-8
> Content-Length: length
> SOAPAction: "http://www.shoppingcartelite.com/AccountSearch"
> 

So, I'm strongly opinionated in them saying. 

> - [Rest API access ](https://api.shoppingcartelite.com/master/sceapi.asmx)for every entity in the system

Because that file really doesn't like like being Rest based!

Anyway, I'll double check with PD team.

**Comment by sergio-eliot-rodriguez** [comment]
hi @stephen485 

thanks for engaging with me. here with a request,

can you please share about what you see if you follow the instructions?
i couldn't easily get an account with SCE:

> To generate your token and access key, go to Setting > Backoffice Setting > API and generate your StoreID,Access Key and Token to use with your calls.

if you share an screenshot, please redact any sensitive info such as the access key and token themselves.

i'm curious to see if they provide a different API endpoint, instead of the ASMX one.

Here's an example of what i'm asking for an app named Acce Mail, you can see there is an api endpoint circled in yellow

![image](https://github.com/PipedreamHQ/pipedream/assets/101526248/4700b6c0-564d-475b-be18-021fe0cd47de)

**Comment by stephen485** [comment]
Here is the API creation screen, but there is no API endpoint shown.
I can ask SCE support for my site endpoint.

***@***.***

hi @stephen485<https://github.com/stephen485>

thanks for engaging with me. here with a request,

can you please share about what you see if you follow the instructions?
i couldn't easily get an account with SCE:

To generate your token and access key, go to Setting > Backoffice Setting > API and generate your StoreID,Access Key and Token to use with your calls.

if you share an screenshot, please redact any sensitive info such as the access key and token themselves.

i'm curious to see if they provide a different API endpoint, instead of the ASMX one.

Here's an example of what i'm asking for an app named Acce Mail, you can see there is an api endpoint circled in yellow

[image]<https://user-images.githubusercontent.com/101526248/257620842-4700b6c0-564d-475b-be18-021fe0cd47de.png>

**Comment by sergio-eliot-rodriguez** [comment]
Yes, please ask support and see where they point us. I checked with the team and we think SCE's API isn't Rest based.

![image](https://github.com/PipedreamHQ/pipedream/assets/101526248/26ced58d-02d7-4981-a7a2-9c1afcbd6e5b)

**Comment by stephen485** [comment]
If this is not REST, I can see if SCE can set up a data feed with new and updated subscriber table rows.
Here is the endpoint link.
https://api.shoppingcartelite.com/master/sceapi.asmx
________________________________

**Comment by sergio-eliot-rodriguez** [comment]
sounds like this could be feasible to your use case right? but this wouldn't imply involving Pipedream, or do you see i.e Pipedream workflows being triggered by new/updated subscriber table rows?

**Comment by stephen485** [comment]
Since SOAP is a no-go and REST is not available, then feeds from SCE to SmartSuite or Retool will work well enough, even though I had hoped to make many SCE workflows to build a dashboard for our fulfillment process; multiple feeds are a more cumbersome approach.
________________________________

**Comment by sergio-eliot-rodriguez** [comment]
noted. thanks for elaborated.
i'll leave the issue open for when SOAP is supported in Pipedream
